lib/kvdb/ikvdb.c

```diff
@@ -1443,10 +1436,12 @@ ikvdb_open(
         goto out;
     }
 
-    err = ikvdb_pmem_only_from_meta(kvdb_home, &meta, &self->ikdb_pmem_only);
-    if (err) {
-        log_errx("cannot open %s", err, kvdb_home);
-        goto out;
+    if (!rdonly_media) {
+        err = ikvdb_pmem_only_from_meta(kvdb_home, &meta, &self->ikdb_pmem_only);
+        if (err) {
+            log_errx("cannot open %s", err, kvdb_home);
+            goto out;
+        }
```

Comment:
Why is `ikvdb_pmem_only_from_media` skipped in this case?

Reply:
ikvdb_pmem_only_from_media() needs to write to the home directory as it creates a file, examines if it could be mmaped with certain flags and then destroys it. This check cannot be done in rdonly/diag modes which could potentially be running on a ROFS or on a home dir with restricted permissions. This check is anyway unneeded in these two open modes as we cannot create new mpool objects (mblock, mdc etc.).